### DISCUSSION TITLE
ZbBridge directly to InfluxDB

### DISCUSSION THREAD

**Discussion by m1k3f15h** [arendst/Tasmota#16417] [Ideas]
Hello Guys,

My actual Situation:
Tasmota ZbBridge publishes sensorvalues (temperature and humidity) on mqtt --> Telegraf put the values into InfluxDB.

I compiled my own tasmota from ZbBridge WITH InfluxDB.
Now my ZbBridge put the Values from the Bridge (RSSI, Heap,...) to InfluxDB, but NOT the Values from my Zigbee Devices.

Did someone else use such a combination?

Greetings from Austria,
Mike

**Comment by arendst** [comment]
I'll have a look how to send the zigbee values to the influxdb driver while keeping the bridge not to busy.

**Comment by arendst** [comment]
As I don't have any zigbe sensor could you provide some logging of your zigbee sensor(s). Preferrable level 3

**Reply by s-hadinger** [reply]
@arendst All the sensor information is sent from `void Z_Device::jsonPublishAttrList(const char * json_prefix, const Z_attribute_list &attr_list, bool include_time) const`

in `xdrv_23_zigbee_2a_devices_impl.ino` line 525

**Reply by arendst** [reply]
Thx. Processing...

**Reply by sfromis** [reply]
Case was also discussed on Discord, where I repeatedly suggested using Zigbee options to make the Zigbee sensor payload the same basic structure as the other sensor data, a list of named objects with a number of values under each. But m1k3f15h did not manage to get to that point. Yet?

My hope was that the current InfluxDB integration would use the JSON format for sensor payloads, but I also expressed a concern that this integration could be piggybacking on the process of JSON building from the normal sensor drivers, and the Zigbee code has a different logic for building the sensor payloads.

One aspect of this is that there is a list of Zigbee options directly affecting how this JSON in structured. IMO "too much" flexibility, but was in part created to mitigate that the original format was "less friendly" to rules and other automation.

This combination of options is what I use, and I have the payload format similarity, meaning that my backend code can handle Zigbee sensors just like wired sensors. This assumes that ZbName has been used to give each device a "friendly name".
```
option cmnd         synonym          value
    83 SetOption83  ZbNameKey        ON
    89 SetOption89  ZbDeviceTopic    OFF
   100 SetOption100 ZbNoPrefix       ON
   101 SetOption101 ZbEndpointSuffix OFF
   110 SetOption110 ZbNoAutoBind     OFF
   112 SetOption112 ZbNameTopic      OFF
   116 SetOption116 ZbNoAutoQuery    OFF
   118 SetOption118 ZbReceivedTopic  OFF
   119 SetOption119 ZbOmitDevice     OFF
   120 SetOption120 ZbEndpointTopic  OFF
```
Not all of the above options are for the JSON format, and some of them are only small non-structural changes.

Here's an example payload with the above set of options:
`{"SNZB-02-TH01":{"Device":"0x2745","Name":"SNZB-02-TH01","Humidity":40.08,"Endpoint":1,"LinkQuality":74}}`

**Reply by arendst** [reply]
Thx. That's useful as the default messages currently won't work.

**Reply by m1k3f15h** [reply]
Hy,
I tested it with "@sfromis" config..
MQT: tele/ZbBridge/SENSOR = {"kg/we":{"Device":"0x2183","Name":"kg/we","Temperature":22.67,"Humidity":65.19,"Pressure":983,"PressureScale":-1,"PressureScaledValue":9839,"SeaPressure":983,"BatteryPercentage":100,"Endpoint":1,"LinkQuality":11}} (retained)

But no entry in InfluxDB :(

mfg
Mike

**Reply by arendst** [reply]
No. That's obvious as there is no link between zigbee and influx at the moment. That's what I'm working on right now.

**Comment by arendst** [comment]
Latest dev adds influxdb sensor support to zigbee binaries with preferred ``SetOption83 1`` and ``SetOption100 1``

Pls do not forget to enable influxdb sensor logging with ``IfxSensor 1`` (as I did and was debugging why I didn't see anything)

Available from OTA download in an hour.

BTW for influxdb debugging enable undocumented and non persistent command ``IfxLog 2``.

**Reply by sfromis** [reply]
The experiences of m1k3f15h suggests that you need other options being off too, to get to the preferred format.

**Reply by arendst** [reply]
OK. Makes sense as mine were all off anyway. So set it to your suggested values (See above).

**Comment by arendst** [comment]
Luckily there is no difference between SO100 0 and SO100 1:
```
SO100 1:
15:37:22.322 IFX: Process {"Switch3":{"Device":"0xAB67","Name":"Switch3","Power":1,"Endpoint":1,"LinkQuality":196}}
15:37:22.327 IFX: Sending
power,device=zigbeepro,sensor=switch3 value=1
endpoint,device=zigbeepro,sensor=switch3 value=1
linkquality,device=zigbeepro,sensor=switch3 value=196
15:37:22.393 MQT: tele/zigbeepro/SENSOR = {"Switch3":{"Device":"0xAB67","Name":"Switch3","Power":1,"Endpoint":1,"LinkQuality":196}}

SO100 0:
15:51:32.980 IFX: Process {"ZbReceived":{"Switch3":{"Device":"0xAB67","Name":"Switch3","Power":0,"Endpoint":1,"LinkQuality":211}}}
15:51:32.985 IFX: Sending
power,device=zigbeepro,sensor=switch3 value=0
endpoint,device=zigbeepro,sensor=switch3 value=1
linkquality,device=zigbeepro,sensor=switch3 value=211
15:51:33.062 MQT: tele/zigbeepro/SENSOR = {"ZbReceived":{"Switch3":{"Device":"0xAB67","Name":"Switch3","Power":0,"Endpoint":1,"LinkQuality":211}}}
```

**Reply by s-hadinger** [reply]
Magic!

**Comment by m1k3f15h** [comment]
Hello,

It Works :)
Thanks a lot!! I love it :)

**Comment by m1k3f15h** [comment]
Is this feature now implemented in the latest master-branch?

mike

**Reply by barbudor** [reply]
It's part of the Releases (master) branch since 1.2.1.1 on Sep 9th 2022

But there's really no reason to focus on Release vs Development

**Reply by sfromis** [reply]
Yes. If what you really mean is if it is included in the ZbBridge release build, that should also be a "Yes".

**Reply by naltatis** [reply]
When I see it correctly influxdb support is in master but there's not official build with influx support builtin and I'd have to do a custom build, right?
https://tasmota.github.io/docs/BUILDS/

**Reply by sfromis** [reply]
You originally asked specifically about the ZbBridge build, and I already confirmed it to be included there, which means both ESP32 and ESP8266.

Other ESP8266 precompiled builds does not include the support, but ESP32 builds does by default.

**Reply by barbudor** [reply]
@naltatis The BUILDs table is nice but not automatically generated, so it may not be fully accurate
The most accurate way is to search for not-commented `#define USE_xxxxx` in the 3 files:
- my_config.h => if it's there, it's in tasmota.bin and tasmota32.bin
- tasmota_configurations.h => official variants for tasmota ESP8266
- tasmota_conifgurations_ESP32.h => official variants for tasmota ESP32 (including C3/S2/S3)

You can also look at the unofficials test builds from Jason : [binaries here](https://github.com/Jason2866/Tasmota-specials/tree/firmware/firmware) and the [configuration files here](https://github.com/Jason2866/Tasmota-build/blob/special_builder/tasmota/user_config_override.h)

**Reply by sfromis** [reply]
The most direct way is to run decode-config or decode-status, based on a device with the binary flashed, this will detail `USE_xxxxxx` flags for features included (not suboptions within features).
